controllers/runnerpool_controller.go

```diff
@@ -65,20 +68,8 @@ func (r *RunnerPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
-	if rp.ObjectMeta.DeletionTimestamp.IsZero() {
-		if !controllerutil.ContainsFinalizer(rp, runnerPoolFinalizer) {
-			err := r.addFinalizer(ctx, rp)
-			if err != nil {
-				log.Error(err, "failed to add finalizer")
-				return ctrl.Result{}, err
-			}
-			// Result does not change even if not requeued here.
-			// This just breaks down one reconciliation loop into small steps for simplicity.
-			log.Info("added finalizer")
-			return ctrl.Result{Requeue: true}, nil
-		}
-	} else {
-		if controllerutil.ContainsFinalizer(rp, runnerPoolFinalizer) {
+	if !rp.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
Kubernetes people like `!= nil`.
```suggestion
	if !rp.ObjectMeta.DeletionTimestamp != nil {
```